### ISSUE TITLE
Bug: Comet from Venus mc reduction should be optional

### ISSUE THREAD

**Issue by vsrisuknimit** [bafolts/terraforming-mars#2954]
Comet from Venus mc reduction should be optional, in case you are Mons or you think the opponent has Law Suits.

**Comment by vsrisuknimit** [comment]
This issue is also a problem for all attack cards that remove resources such as Air Raid, Sabotage, and Hire Raider. They all should provide an option to remove from the resource from no one.

We may need to modify RemoveAnyPlants.ts from DeferredAction or make another class based on it to provide an option to skip attacking.

**Comment by nwai90** [comment]
Checked with @ssimeonoff , Comet From Venus is similar mechanic to Flooding. We can return an `OrOptions` instead of a `SelectPlayer` for that card only.

For all other attack cards the effect must be performed and is not optional, only removing plants is.

https://ssimeonoff.github.io/cards-list#218#188

**Comment by vsrisuknimit** [comment]
![image](https://user-images.githubusercontent.com/14239220/111400821-7de44a00-869e-11eb-89fc-159ef7a6cfac.png)
http://www.fryxgames.se/TerraformingMars/TMRULESFINAL.pdf

**Comment by ssimeonoff** [comment]
Most of the attack cards says "may" or "up to", but some don't.

**Comment by vsrisuknimit** [comment]
@ssimeonoff I am pretty sure those are oversights from the designers. The rule on the red-border cube should apply.

Let me know if you have an example.

**Comment by ssimeonoff** [comment]
This means that Predators and Ants can be activated targeting an opponent, but removing 0?

**Comment by Lynesth** [comment]
@ssimeonoff Nope, because it's a cost. There was a discussion about it already and confirmation by Fryx, a bit too lazy to find it now.

**Comment by vsrisuknimit** [comment]
@ssimeonoff  See https://boardgamegeek.com/thread/2374854/predators-and-ants-clarification-please.

As @Lynesth said, it is a cost (before a red arrow).

The question is whether card text overwrite the rulebook for these cards: https://ssimeonoff.github.io/cards-list#063#T02#188#C02. Or was it an oversight in the card text.

Can you Air Raid without stealing? Flood remove less than 4 mc? Mining Expedition and remove one plant? Banned delegate and not remove any delegate?

Not sure about Banned Delegate, but the general rules on red-border non-production non-cost should applies to these cards. I think so.
